resources/idlharness.js

```diff
@@ -3288,15 +3292,21 @@ function idl_test(srcs, deps, idl_setup_func) {
         deps = (deps instanceof Array) ? deps : [deps] || [];
         var setup_error = null;
         return Promise.all(
-            srcs.concat(deps).map(function(spec) {
-                return fetch_spec(spec);
-            }))
-            .then(function(idls) {
+            srcs.concat(deps).map(fetch_spec))
+            .then(function(results) {
+                const astArray = results.map(result =>
+                    WebIDL2.parse(result.idl, { sourceName: result.spec })
+                );
+                const validations = WebIDL2.validate(astArray);
+                if (validations.length) {
+                    const message = validations.map(v => v.message).join("\n\n");
+                    throw new Error(message);
```

Comment:
I think that we should fall back to current behaviour (of ignoring invalid syntax, and testing what we can nonetheless).
```suggestion
                    setup_error = setup_error || new Error(message);
```